root/user/UserCollections.js

```diff
@@ -0,0 +1,199 @@
+import {withCatalystContext} from '../context';
+import {unwrapNl} from '../static/scripts/common/i18n';
+import {formatPluralEntityTypeName}
+  from '../static/scripts/common/utility/formatEntityTypeName';
+import {
+  defineNameColumn,
+  typeColumn,
+  subscriptionColumn,
+  defineActionsColumn,
+} from '../utility/tableColumns';
+
+type Props = {|
+  +$c: {user: EditorT} & CatalystContextT,
+  +collaborativeCollections: CollectionListT,
+  +ownCollections: CollectionListT,
+  +user: EditorT,
+|};
+
+type CollectionListT = {|
+  +[string]: $ReadOnlyArray<CollectionT>,
+|};
+
+const collectionsListTitle = {
+  area: N_l('Area Collections'),
+  artist: N_l('Artist Collections'),
+  event: N_l('Event Collections'),
+  instrument: N_l('Instrument Collections'),
+  label: N_l('Label Collections'),
+  place: N_l('Place Collections'),
+  recording: N_l('Recording Collections'),
+  release: N_l('Release Collections'),
+  release_group: N_l('Release Group Collections'),
+  series: N_lp('Series Collections', 'plural'),
+  work: N_l('Work Collections'),
+};
+
+function formatCollaboratorNumber(
+  collaborators: $ReadOnlyArray<EditorT>,
+  currentUser: EditorT | null,
+) {
+  const isCollaborator = !!currentUser && some(
+    collaborators,
+    (collaborator => collaborator.id === currentUser.id),
+  );
+
+  return isCollaborator ? (
+    exp.l('{collaborator_number} (including you)',
+          {collaborator_number: collaborators.length})
+  ) : (
+    collaborators.length
+  );
+}
+
+function formatPrivacy(isPublic: boolean) {
+  return isPublic ? l('Public') : l('Private');
+}
+
+function collaborativePrivacy(
+  collection: CollectionT,
+  currentUser: EditorT | null,
+) {
+  return !!currentUser && !!collection.editor &&
+    collection.editor.id === currentUser.id ? (
+      formatPrivacy(collection.public) + ' ' + l('(your collection)')
+    ) : (
+      formatPrivacy(collection.public)
+    );
+}
+
+const CollectionsEntitySection = ({
+  $c,
+  isCollaborative,
+  collections,
+  type,
+  user,
+}) => {
+  const viewingOwnProfile = !!$c.user && $c.user.id === user.id;
+  const nameColumn = defineNameColumn(N_l('Collection'));
+  const sizeColumn = {
+    Header: () => formatPluralEntityTypeName(type),
+    accessor: 'entity_count',
+    id: 'size',
+  };
+  const collaboratorsColumn = {
+    Cell: ({cell: {value}}) => formatCollaboratorNumber(value, $c.user),
+    Header: N_l('Collaborators'),
+    accessor: 'collaborators',
+    id: 'collaborators',
+  };
+  const privacyColumn = {
+    Cell: ({row: {original}}) => collaborativePrivacy(original, $c.user),
+    Header: N_l('Privacy'),
+    accessor: 'public',
+    id: 'privacy',
+  };
+  const actionsColumn = defineActionsColumn([
+    [l('Edit'), '/edit'],
+    [l('Remove'), '/delete'],
+  ]);
+  const columns = React.useMemo(
+    () => [
+      nameColumn,
+      typeColumn,
+      sizeColumn,
+      collaboratorsColumn,
+      ...viewingOwnProfile ? [subscriptionColumn] : [],
```

Comment:
Parentheses would make the actual spread operation here much easier to parse.
```suggestion
      ...(viewingOwnProfile ? [subscriptionColumn] : []),
```

Reply:
Fixed in https://github.com/metabrainz/musicbrainz-server/pull/1246/commits/6ee5d3d424b1dda8aecefe6be1b9f7d2ac4092e1#diff-3f513fa6fa1fed40110c8c20d602c01fR116-R118